feat: add `succinct-zkvm` target

nhtyy opened the pull request:
<!-- homu-ignore:start -->
<!--
If this PR is related to an unstable feature or an otherwise tracked effort,
please link to the relevant tracking issue here. If you don't know of a related
tracking issue or there are none, feel free to ignore this.

This PR will get automatically assigned to a reviewer. In case you would like
a specific user to review your work, you can assign it to them by using

    r? <reviewer name>
-->
<!-- homu-ignore:end -->

Adds a new riscv32im target with a new os (`succinct-zkvm`). 

Previously, there was already a `zkvm` target which is in the process of being removed, due to [this issue](https://github.com/rust-lang/rust/issues/135376), which we think is the right call.

This PR DOES NOT attempt to resolve the aforementioned issue for the risc0 target.

rust-log-analyzer commented:
The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/13823418432/job/38673751419) [(plain)](https://github.com/rust-lang/rust/commit/67c45e3d0d1690831fd8d90b7f971131e75eb451/checks/38673751419/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
info: removing rustup binaries
info: rustup is uninstalled
##[group]Image checksum input
mingw-check-tidy
# We use the ghcr base image because ghcr doesn't have a rate limit
# and the mingw-check-tidy job doesn't cache docker images in CI.
FROM ghcr.io/rust-lang/ubuntu:22.04

ARG DEBIAN_FRONTEND=noninteractive
RUN apt-get update && apt-get install -y --no-install-recommends \
  g++ \
  make \
---

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
#
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#12 2.911 Building wheels for collected packages: reuse
#12 2.912   Building wheel for reuse (pyproject.toml): started
#12 3.129   Building wheel for reuse (pyproject.toml): finished with status 'done'
#12 3.130   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=0c2fd2aaf7b0bf8d6e131220aff14712a774c2ca462f3204d25460cbcf610b63
#12 3.131   Stored in directory: /tmp/pip-ephem-wheel-cache-zdvfypv2/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#12 3.133 Successfully built reuse
#12 3.134 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#12 3.558 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#12 3.558 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#12 4.108 Collecting virtualenv
#12 4.160   Downloading virtualenv-20.29.3-py3-none-any.whl (4.3 MB)
#12 4.368      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 20.9 MB/s eta 0:00:00
#12 4.427 Collecting platformdirs<5,>=3.9.1
#12 4.436   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#12 4.460 Collecting distlib<1,>=0.3.7
#12 4.468   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#12 4.482      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 37.8 MB/s eta 0:00:00
#12 4.523 Collecting filelock<4,>=3.12.2
#12 4.531   Downloading filelock-3.17.0-py3-none-any.whl (16 kB)
#12 4.612 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#12 4.799 Successfully installed distlib-0.3.9 filelock-3.17.0 platformdirs-4.3.6 virtualenv-20.29.3
#12 4.799 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#12 DONE 4.9s

#13 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#13 DONE 0.0s
---
extracting /checkout/obj/build/cache/2025-02-18/rust-std-beta-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/2025-02-18/rustc-beta-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/2025-02-18/cargo-beta-x86_64-unknown-linux-gnu.tar.xz
##[group]Building bootstrap
    Updating git repository `https://github.com/nhtyy/cc-rs`
    Updating crates.io index
---
   Compiling version_check v0.9.5
   Compiling generic-array v0.14.7
   Compiling libc v0.2.167
   Compiling shlex v1.3.0
   Compiling cc v1.2.0 (https://github.com/nhtyy/cc-rs?branch=n%2Fadd-succinct-target#424acb08)
   Compiling quote v1.0.37
   Compiling syn v2.0.87
   Compiling aho-corasick v1.1.3
   Compiling serde v1.0.215
   Compiling crossbeam-utils v0.8.20
---
DirectMap4k:      149440 kB
DirectMap2M:     8239104 kB
DirectMap1G:    10485760 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
##[group]Building bootstrap
    Finished `dev` profile [unoptimized] target(s) in 0.05s
##[endgroup]
WARN: currently no CI rustc builds have rustc debug assertions enabled. Please either set `rust.debug-assertions` to `false` if you want to use download CI rustc or set `rust.download-rustc` to `false`.
downloading https://static.rust-lang.org/dist/2025-02-18/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
---
##[endgroup]
fmt check
Diff in /checkout/library/test/src/lib.rs:555:
 
     // Emscripten can catch panics but other wasm targets cannot
     let ignore_because_no_process_support = desc.should_panic != ShouldPanic::No
-        && (cfg!(target_family = "wasm") || cfg!(target_os = "zkvm") || cfg!(target_os = "succinct-zkvm"))
+        && (cfg!(target_family = "wasm")
+            || cfg!(target_os = "zkvm")
+            || cfg!(target_os = "succinct-zkvm"))
         && !cfg!(target_os = "emscripten");
 
     if force_ignore || desc.ignore || ignore_because_no_process_support {
Diff in /checkout/library/test/src/console.rs:316:
     // Prevent the usage of `Instant` in some cases:
     // - It's currently not supported for wasm targets without Emscripten nor WASI.
     // - It's currently not supported for zkvm targets.
-    let is_instant_unsupported =
-        (cfg!(target_family = "wasm") && cfg!(target_os = "unknown")) || cfg!(target_os = "zkvm") || cfg!(target_os = "succinct-zkvm");
+    let is_instant_unsupported = (cfg!(target_family = "wasm") && cfg!(target_os = "unknown"))
+        || cfg!(target_os = "zkvm")
+        || cfg!(target_os = "succinct-zkvm");
 
     let start_time = (!is_instant_unsupported).then(Instant::now);
     run_tests(opts, tests, |x| on_test_event(&x, &mut st, &mut *out))?;
Diff in /checkout/library/std/src/sys/random/zkvm.rs:25:
 pub fn fill_bytes(bytes: &mut [u8]) {
     unsafe { abi::sys_rand(bytes.as_mut_ptr(), bytes.len()) };
 }
+
Diff in /checkout/library/std/src/sys/pal/zkvm/abi.rs:1:
 //! ABI definitions for symbols exported by <_>-zkvm targets.
-//! 
+//!
 //! Ideally, these should be the minimum viable symbols to support the std-lib.
 
 #![allow(dead_code)]
Diff in /checkout/library/std/src/sys/pal/zkvm/os.rs:1:
-use super::unsupported;
-use crate::error::Error as StdError;
-use crate::ffi::{OsStr, OsString};
-use crate::marker::PhantomData;
-use crate::path::{self, PathBuf};
-use crate::{fmt, io};
-
 #[cfg(target_vendor = "risc0")]
 use {
     super::{WORD_SIZE, abi},
Diff in /checkout/library/std/src/sys/pal/zkvm/os.rs:11:
     crate::sys::os_str,
     crate::sys_common::FromInner,
 };
+
+use super::unsupported;
---
+use crate::{fmt, io};
 
 pub fn errno() -> i32 {
     0
Diff in /checkout/library/std/src/sys/pal/zkvm/args.rs:1:
-use crate::ffi::OsString;
-use crate::fmt;
-
 #[cfg(target_vendor = "risc0")]
 use {
     super::{WORD_SIZE, abi},
Diff in /checkout/library/std/src/sys/pal/zkvm/args.rs:7:
-    crate::sys_common::FromInner,
     crate::sys::os_str,
+    crate::sys_common::FromInner,
 };
+
+use crate::ffi::OsString;
+use crate::fmt;
 
 pub struct Args {
     i_forward: usize,
Diff in /checkout/src/bootstrap/src/core/sanity.rs:35:
 const STAGE0_MISSING_TARGETS: &[&str] = &[
     // just a dummy comment so the list doesn't get onelined
     "wasm32-wali-linux-musl",
-    "riscv32im-succinct-zkvm-elf"
+    "riscv32im-succinct-zkvm-elf",
 ];
 
 /// Minimum version threshold for libstdc++ required when using prebuilt LLVM
fmt: checked 5898 files
Build completed unsuccessfully in 0:01:00
```

</details>

jieyouxu commented on src/llvm-project:
Is this submodule bump intentional?

nhtyy replied:
fixed!

jieyouxu commented on src/bootstrap/Cargo.toml:
```diff
@@ -39,7 +39,11 @@ test = false
 #
 # Do not upgrade this crate unless https://github.com/rust-lang/cc-rs/issues/1317 is fixed.
-cc = "=1.1.22"
+#
+# todo(succinct): remove this change that was explicity against the rules.
+# This just happens to work, and ideally cc-rs will fix the target parsing issue before our target is merged.
+cc = "=1.2.0"
+# cc = "=1.1.22"
```
Problem: you cannot bump bootstrap's `cc` here in this PR. cc > 1.1.22 still has problems with target parsing issue.

nhtyy replied:
Left some comments above in the changes, it just so happened that I had forked that version of `cc-rs` and it worked. Happy to redo it with the correct version if its blocking on review

jieyouxu replied:
It's fine to experiment with a forked `cc` version if you need it for testing under PR CI or whatever, but we can't merge this into actual `master` until this bootstrap `cc` bump is reverted.

jieyouxu replied:
EDIT: ah I see, it's not blocking for review, only blocking for approval/merge.

jieyouxu commented on src/bootstrap/Cargo.lock:
```diff
@@ -88,9 +88,8 @@ dependencies = [
-version = "1.1.22"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "9540e661f81799159abee814118cc139a2004b3a3aa3ea37724a1b66530b90e0"
+version = "1.2.0"
+source = "git+https://github.com/nhtyy/cc-rs?branch=n%2Fadd-succinct-target#424acb084e212c330de96e06cab69be0d9df55f4"
```
Problem: this is also pointing to a custom `cc`-rs fork that I don't think we can land upstream here in rust-lang/rust.

jieyouxu replied:
(Same review comment chain as the bootstrap `cc` version).

jieyouxu commented:
I would recommend splitting this PR up into at least two PRs:

1. First PR adds an `no_std` minimal compiler target definition and roll a compiler lead reviewer via `r? compiler_leads`.
2. Second PR adds the relevant std support.

This is to make it easier to review because it's hard to review the mixed compiler/library changes in this one PR (compiler / library are maintained by different teams).

nhtyy commented:
@jieyouxu TYSM for getting to this so quickly.

My plan here, is that given the volatile state of the `zkvm` (issue in description), I was expecting this to not be merged till that was fixed.

Do you think this OS should live in its own `pal` folder? I currently just shared with risc0 because there is alot of overlap. If thats the case I think creating a `pal/succinct-zkvm` will clean this PR up a lot and make reviewing much easier. Ideally, avoiding the need for two PRs.

jieyouxu commented:
> My plan here, is that given the volatile state of the `zkvm` (issue in description), I was expecting this to not be merged till that was fixed.

Ah. If this is *another* `zkvm` that's different enough than the other `zkvm` targets (in terms of the zkvm implementaiton), than IMO a separate `target_os` is definitely more faithful than just lumping all the different `zkvm` targets under one `target_os = "zkvm"`. After all, what use is sharing a `target_os` if the underlying  in reality substantially different and incompatible with each other? Maybe even `target_os = "succinct"` (I don't know what would be a good `target_os` here for your target)?

And yeah, sorting out the current zkvm target situation would be ideal to avoid confusion and footguns in the long term.

> Do you think this OS should live in its own `pal` folder? I currently just shared with risc0 because there is alot of overlap. If thats the case I think creating a `pal/succinct-zkvm` will clean this PR up a lot and make reviewing much easier. Ideally, avoiding the need for two PRs.

I can't really comment on this because I'm not a library maintainer/reviewer. This is partly why why I recommended doing the library parts in a separate PR, so that you can discuss the library implementation decisions with a library reviewer.

nhtyy committed:
feat: succinct target

nhtyy commented:
Ok went ahead and moved everything to not rely on (mostly) any `zkvm` code.

The PR is definitely bigger, so I agree with your suggestion, Ill go ahead and split up these PRs soon.

rust-log-analyzer commented:
The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/13824177613/job/38676015720) [(plain)](https://github.com/rust-lang/rust/commit/5c1dbd3dc07ff240f5e819190ea0d713ab8b765e/checks/38676015720/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
info: removing rustup binaries
info: rustup is uninstalled
##[group]Image checksum input
mingw-check-tidy
# We use the ghcr base image because ghcr doesn't have a rate limit
# and the mingw-check-tidy job doesn't cache docker images in CI.
FROM ghcr.io/rust-lang/ubuntu:22.04

ARG DEBIAN_FRONTEND=noninteractive
RUN apt-get update && apt-get install -y --no-install-recommends \
  g++ \
  make \
---

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
#
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#12 2.944 Building wheels for collected packages: reuse
#12 2.945   Building wheel for reuse (pyproject.toml): started
#12 3.159   Building wheel for reuse (pyproject.toml): finished with status 'done'
#12 3.160   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=0c2fd2aaf7b0bf8d6e131220aff14712a774c2ca462f3204d25460cbcf610b63
#12 3.160   Stored in directory: /tmp/pip-ephem-wheel-cache-6r1ie3zt/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#12 3.162 Successfully built reuse
#12 3.163 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#12 3.560 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#12 3.560 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#12 4.098 Collecting virtualenv
#12 4.153   Downloading virtualenv-20.29.3-py3-none-any.whl (4.3 MB)
#12 4.306      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 28.6 MB/s eta 0:00:00
#12 4.370 Collecting filelock<4,>=3.12.2
#12 4.380   Downloading filelock-3.17.0-py3-none-any.whl (16 kB)
#12 4.418 Collecting platformdirs<5,>=3.9.1
#12 4.426   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#12 4.448 Collecting distlib<1,>=0.3.7
#12 4.456   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#12 4.469      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 41.5 MB/s eta 0:00:00
#12 4.550 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#12 4.731 Successfully installed distlib-0.3.9 filelock-3.17.0 platformdirs-4.3.6 virtualenv-20.29.3
#12 4.731 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#12 DONE 4.8s

#13 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#13 DONE 0.0s
---
extracting /checkout/obj/build/cache/2025-02-18/rust-std-beta-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/2025-02-18/rustc-beta-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/2025-02-18/cargo-beta-x86_64-unknown-linux-gnu.tar.xz
##[group]Building bootstrap
    Updating git repository `https://github.com/nhtyy/cc-rs`
    Updating crates.io index
---
   Compiling version_check v0.9.5
   Compiling generic-array v0.14.7
   Compiling libc v0.2.167
   Compiling shlex v1.3.0
   Compiling cc v1.2.0 (https://github.com/nhtyy/cc-rs?branch=n%2Fadd-succinct-target#424acb08)
   Compiling quote v1.0.37
   Compiling syn v2.0.87
   Compiling aho-corasick v1.1.3
   Compiling regex-syntax v0.8.5
   Compiling crossbeam-utils v0.8.20
---
DirectMap4k:      131008 kB
DirectMap2M:     7208960 kB
DirectMap1G:    11534336 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
##[group]Building bootstrap
    Finished `dev` profile [unoptimized] target(s) in 0.05s
##[endgroup]
WARN: currently no CI rustc builds have rustc debug assertions enabled. Please either set `rust.debug-assertions` to `false` if you want to use download CI rustc or set `rust.download-rustc` to `false`.
downloading https://static.rust-lang.org/dist/2025-02-18/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
---
     // Prevent the usage of `Instant` in some cases:
     // - It's currently not supported for wasm targets without Emscripten nor WASI.
     // - It's currently not supported for zkvm targets.
-    let is_instant_unsupported =
-        (cfg!(target_family = "wasm") && cfg!(target_os = "unknown")) || cfg!(target_os = "zkvm") || cfg!(target_os = "succinct-zkvm");
+    let is_instant_unsupported = (cfg!(target_family = "wasm") && cfg!(target_os = "unknown"))
+        || cfg!(target_os = "zkvm")
+        || cfg!(target_os = "succinct-zkvm");
 
     let start_time = (!is_instant_unsupported).then(Instant::now);
     run_tests(opts, tests, |x| on_test_event(&x, &mut st, &mut *out))?;
Diff in /checkout/library/test/src/lib.rs:555:
 
     // Emscripten can catch panics but other wasm targets cannot
     let ignore_because_no_process_support = desc.should_panic != ShouldPanic::No
-        && (cfg!(target_family = "wasm") || cfg!(target_os = "zkvm") || cfg!(target_os = "succinct-zkvm"))
+        && (cfg!(target_family = "wasm")
+            || cfg!(target_os = "zkvm")
+            || cfg!(target_os = "succinct-zkvm"))
         && !cfg!(target_os = "emscripten");
 
     if force_ignore || desc.ignore || ignore_because_no_process_support {
Diff in /checkout/library/std/src/random.rs:48:
 /// VxWorks                | `randABytes` after waiting for `randSecure` to become ready
 /// WASI                   | [`random_get`](https://github.com/WebAssembly/WASI/blob/main/legacy/preview1/docs.md#-random_getbuf-pointeru8-buf_len-size---result-errno)
 /// ZKVM                   | `sys_rand`
-/// SUCCINCT-ZKVM          | `sys_rand` 
+/// SUCCINCT-ZKVM          | `sys_rand`
 ///
 /// Note that the sources used might change over time.
 ///
Diff in /checkout/library/std/src/sys/pal/succint-zkvm/abi.rs:1:
 //! ABI definitions for symbols exported by <_>-zkvm targets.
-//! 
+//!
 //! Ideally, these should be the minimum viable symbols to support the std-lib.
 
 #![allow(dead_code)]
Diff in /checkout/library/std/src/sys/random/zkvm.rs:25:
 pub fn fill_bytes(bytes: &mut [u8]) {
     unsafe { abi::sys_rand(bytes.as_mut_ptr(), bytes.len()) };
 }
+
Diff in /checkout/library/std/src/sys/alloc/mod.rs:7:
 // add fast paths for low alignment values.
 #[allow(dead_code)]
 const MIN_ALIGN: usize = if cfg!(any(
-    all(target_arch = "riscv32", any(target_os = "espidf", target_os = "zkvm", target_os = "succinct-zkvm")),
+    all(
+        target_arch = "riscv32",
+        any(target_os = "espidf", target_os = "zkvm", target_os = "succinct-zkvm")
+    ),
     all(target_arch = "xtensa", target_os = "espidf"),
 )) {
     // The allocator on the esp-idf and zkvm platforms guarantees 4 byte alignment.
Diff in /checkout/library/std/src/sys/random/succinct_zkvm.rs:3:
 pub fn fill_bytes(bytes: &mut [u8]) {
     unsafe { abi::sys_rand(bytes.as_mut_ptr(), bytes.len()) };
 }
+
Diff in /checkout/src/bootstrap/src/core/sanity.rs:35:
 const STAGE0_MISSING_TARGETS: &[&str] = &[
     // just a dummy comment so the list doesn't get onelined
     "wasm32-wali-linux-musl",
-    "riscv32im-succinct-zkvm-elf"
+    "riscv32im-succinct-zkvm-elf",
 ];
 
 /// Minimum version threshold for libstdc++ required when using prebuilt LLVM
fmt: checked 5905 files
Build completed unsuccessfully in 0:01:19
```

</details>

jhpratt commented:
r? compiler

jieyouxu commented:
r? compiler_leads

bors commented:
:umbrella: The latest upstream changes (presumably #138448) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

nhtyy commented:
Closing this in favor of separating the std library impl from adding target.
